transport/src/main/java/io/netty/channel/OwnedIoEventLoop.java

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright 2025 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import io.netty.util.concurrent.AbstractScheduledEventExecutor;
+import io.netty.util.concurrent.DefaultPromise;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.GlobalEventExecutor;
+import io.netty.util.concurrent.OrderedEventExecutor;
+import io.netty.util.concurrent.Promise;
+import io.netty.util.internal.ObjectUtil;
+import io.netty.util.internal.PlatformDependent;
+
+import java.util.Objects;
+import java.util.Queue;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * {@link IoEventLoop} implementation that is owned by the user and so needs to be driven by the user with the given
+ * {@link Thread}. This means that the user is responsible to call either {@link #run()} or {@link #waitAndRun(long)}
+ * to execute IO or tasks that were submitted to this {@link IoEventLoop}.
+ * <p>
+ * This is for <strong>advanced use-cases only</strong>, where the user wants to own the {@link Thread} that drives the
+ * {@link IoEventLoop} to also do other work. That said care must be taken that the {@link #run() or
+ * {@link #waitAndRun()}} methods are called in a timely fashion.
+ */
+public final class OwnedIoEventLoop extends AbstractScheduledEventExecutor
```

Comment:
I am open for a better name

Reply:
IoEventLoopControl since it can be used to drive I/O related operations in a event loop-like way

Reply:
I would prefer to have it end with `IoEventLoop` so its clear to people that this is also the one that you would inject into `Bootstrap` etc.

Reply:
Renamed to `ManualIoEventLoop`

Reply:
@chrisvest @franz1981 how you like the name ?

Reply:
Better! Consider that Martin Thompson's Agents in Agrona used ControlledXYZ but manual is more intuitive to me, so +100